### ISSUE TITLE
ambiguous `ruby-head`

### ISSUE THREAD

**Issue by marcandre** [rvm/rvm#4986]
Took me a while to understand what was going on when I entered `rvm reinstall ruby-head` and it failed with strange errors.

Install failed, and somehow `truffle-ruby` was somehow interferring?

See 10 lines before the end of:
<details>

```
[2020-09-26 15:04:45] __rvm_make
__rvm_make () 
{ 
    \make "$@" || return $?
}
current path: /Users/mal/.rvm/src/ruby-head
GEM_PATH=/Users/mal/.rvm/gems/ruby-2.7.1:/Users/mal/.rvm/gems/ruby-2.7.1@global
GEM_HOME=/Users/mal/.rvm/gems/ruby-2.7.1
PATH=/usr/local/opt/coreutils/bin:/usr/local/opt/pkg-config/bin:/usr/local/opt/libtool/bin:/usr/local/opt/automake/bin:/usr/local/opt/autoconf/bin:/Users/mal/.rvm/gems/ruby-2.7.1/bin:/Users/mal/.rvm/gems/ruby-2.7.1@global/bin:/Users/mal/.rvm/rubies/ruby-2.7.1/bin:/usr/local/opt/postgresql@9.6/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/Applications/VMware Fusion.app/Contents/Public:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/mal/.ec2/api/bin:/Users/mal/ruby/dev/spec/mspec/bin:/Users/mal/.bin:/usr/local/Cellar/node/6.8.1/bin:/usr/local/bin:/usr/local/opt/fzf/bin:/Users/mal/.rvm/bin
command(2): __rvm_make -j4
++ make -j4
	BASERUBY = /Users/mal/.rvm/wrappers/truffleruby-20.0.0/ruby --disable=gems
	CC = gcc -fdeclspec
	LD = ld
	LDSHARED = gcc -fdeclspec -dynamiclib
	CFLAGS = -O3 -ggdb3 -Wall -Wextra -Werror=deprecated-declarations -Werror=division-by-zero -Werror=implicit-function-declaration -Werror=implicit-int -Werror=pointer-arith -Werror=shorten-64-to-32 -Werror=write-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Werror=extra-tokens -std=gnu99  -fno-common -pipe 
	XCFLAGS = -D_FORTIFY_SOURCE=2 -fstack-protector-strong -fno-strict-overflow -DRUBY_DEVEL=1 -fvisibility=hidden -DRUBY_EXPORT -DCANONICALIZATION_FOR_MATHN -I. -I.ext/include/x86_64-darwin18 -I./include -I. -I./enc/unicode/12.1.0
	CPPFLAGS = -I/usr/local/opt/libyaml/include -I/usr/local/opt/libksba/include -I/usr/local/opt/readline/include -I/usr/local/opt/zlib/include -I/usr/local/opt/openssl@1.1/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   
	DLDFLAGS = -Wl,-undefined,dynamic_lookup -Wl,-multiply_defined,suppress -L/usr/local/opt/libyaml/lib -L/usr/local/opt/libksba/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/zlib/lib -L/usr/local/opt/openssl@1.1/lib -install_name /Users/mal/.rvm/rubies/ruby-head/lib/libruby.3.0.dylib -compatibility_version 3.0 -current_version 3.0.0  -fstack-protector-strong -framework Security -framework Foundation  -fstack-protector-strong -framework Security -framework Foundation  
	SOLIBS = -lpthread -lgmp -ldl -lobjc
	LANG = en_CA.UTF-8
	LC_ALL = 
	LC_CTYPE = 
	MFLAGS = - --jobserver-fds=4,5 -j
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
compiling ./main.c
compiling dmydln.c
generating id.h
generating miniprelude.c
compiling dmyext.c
making array.rbinc
/Users/mal/.rvm/wrappers/truffleruby-20.0.0/ruby --disable=gems ./tool/gen_dummy_probes.rb ./probes.d > probes.dmyh
translating probes probes.d
. ./vm_opts.h
making ast.rbinc
/Users/mal/.rvm/src/ruby-head/tool/ruby_vm/helpers/c_escape.rb:15:in `const_missing': uninitialized constant RubyVM (NameError)
	from /Users/mal/.rvm/src/ruby-head/tool/ruby_vm/helpers/c_escape.rb:15:in `<top (required)>'
	from ./tool/mk_builtin_loader.rb:5:in `require_relative'
	from ./tool/mk_builtin_loader.rb:5:in `<main>'
make: *** [array.rbinc] Error 1
make: *** Waiting for unfinished jobs....
make: *** [id.h] Error 1
/Users/mal/.rvm/src/ruby-head/tool/ruby_vm/helpers/c_escape.rb:15:in `const_missing': uninitialized constant RubyVM (NameError)
	from /Users/mal/.rvm/src/ruby-head/tool/ruby_vm/helpers/c_escape.rb:15:in `<top (required)>'
	from ./template/prelude.c.tmpl:9:in `require_relative'
	from ./template/prelude.c.tmpl:9:in `<main>'
	from /Users/mal/.rvm/rubies/truffleruby-20.0.0/lib/mri/erb.rb:901:in `eval'
	from /Users/mal/.rvm/rubies/truffleruby-20.0.0/lib/mri/erb.rb:901:in `result'
	from ./tool/generic_erb.rb:41:in `block (2 levels) in <main>'
	from ./tool/generic_erb.rb:41:in `call'
	from ./tool/generic_erb.rb:41:in `block in <main>'
	from ./tool/generic_erb.rb:34:in `map'
	from ./tool/generic_erb.rb:34:in `<main>'
/Users/mal/.rvm/src/ruby-head/tool/ruby_vm/helpers/c_escape.rb:15:in `const_missing': uninitialized constant RubyVM (NameError)
	from /Users/mal/.rvm/src/ruby-head/tool/ruby_vm/helpers/c_escape.rb:15:in `<top (required)>'
	from ./tool/mk_builtin_loader.rb:5:in `require_relative'
	from ./tool/mk_builtin_loader.rb:5:in `<main>'
make: *** [ast.rbinc] Error 1
make: *** [miniprelude.c] Error 1
++ return 2
```
</details>

First I thought it was trying to reinstall truffleruby, and next I did...

```sh
$ rvm uninstall ruby-head
truffleruby-head - #removing src/truffleruby-head..
truffleruby-head - #removing rubies/truffleruby-head..
$ rvm uninstall ruby-head
jruby-head - #removing src/jruby-head..
jruby-head - #removing rubies/jruby-head..
$ rvm uninstall ruby-head
ruby-head - #removing src/ruby-head..
```

??? Seems to me like `ruby-head` should never resolve to truffleruby/jruby and always to ruby (i.e. MRI)

It still wouldn't install `ruby-head`; I had to remove `truffleruby-20.0.0` to be able to successfully install; I only got a small error at the end: `mkdir: /Users/mal/.rvm/gems/ruby-head@global/bin: Not a directory`

**Comment by BrianHawley** [comment]
@marcandre this is fixed in PR #5116.

**Comment by marcandre** [comment]
🎉
